lib/screens/first_screen.dart

```diff
@@ -0,0 +1,77 @@
+import 'dart:io';
+import 'package:flutter/material.dart';
+import 'package:navigation/screens/second_screen.dart';
+
+class FirstScreen extends StatefulWidget {
+  const FirstScreen({Key? key}) : super(key: key);
+
+  @override
+  FirstScreenState createState() => FirstScreenState();
+}
+
+class FirstScreenState extends State<FirstScreen> {
+
+  @override
+  Widget build(BuildContext context) {
+    return WillPopScope(
+      onWillPop: () => showExitDialog(context),
+      child: Scaffold(
+          appBar: AppBar(
+            title: const Text("First screen"),
+          ),
+          body: Center(
+            child: ElevatedButton(
+              child: const Text("Go next"),
+              onPressed: () {
+                navigateAndGetData(context);
+              },
+            ),
+          )),
+    );
+  }
+
+  void navigateAndGetData(BuildContext context) async {
+    final result = await Navigator.push(
+        context, MaterialPageRoute(builder: (context) => const SecondScreen()));
+    showDialog(
+        context: context,
+        builder: (context) => getAlertDialog(result, context));
```

Comment:
thank you for notice. In context of this application this piece of code must work anyway, because from second screen transfer not null data. But in your case in method Widget getAlertDialog(Object result, BuildContext context) you can change type Object to Object? . End even provide default data if result from second screen is null.